File: app/src/main/java/com/chad/baserecyclerviewadapterhelper/entity/MultipleItem.java
package com.chad.baserecyclerviewadapterhelper.entity;

import com.chad.library.adapter.base.entity.MultiItemEntity;

public class MultipleItem implements MultiItemEntity {

  public static final int TEXT = 1;
  public static final int IMG = 2;
  public static final int IMG_TEXT = 3;
  public static final int TEXT_SPAN_SIZE = 3;
  public static final int IMG_SPAN_SIZE = 1;
  public static final int IMG_TEXT_SPAN_SIZE = 4;
  public static final int IMG_TEXT_SPAN_SIZE_MIN = 2;
  private int itemType;
  private int spanSize;

  public MultipleItem(int itemType, int spanSize, String content) {
    this.itemType = itemType;
    this.spanSize = spanSize;
    this.content = content;
  }

  public MultipleItem(int itemType, int spanSize) {
    this.itemType = itemType;
    this.spanSize = spanSize;
  }

  public int getSpanSize() {
    return spanSize;
  }

  public void setSpanSize(int spanSize) {
    this.spanSize = spanSize;
  }

  private String content;

  public String getContent() {
    return content;
  }

  public void setContent(String content) {
    this.content = content;
  }

  @Override
  public int getItemType() {
    return itemType;
  }
}

File: kotlinapp/src/main/java/com/allen/kotlinapp/entity/MultipleItem.kt
package com.allen.kotlinapp.entity

import com.chad.library.adapter.base.entity.MultiItemEntity

/**
 * 文 件 名: MultipleItem
 * 创 建 人: Allen
 * 创建日期: 2017/6/13 14:20
 * 修改时间：
 * 修改备注：
 */
class MultipleItem : MultiItemEntity {
    private var itemType: Int = 0
    var spanSize: Int = 0

    constructor(itemType: Int, spanSize: Int, content: String) {
        this.itemType = itemType
        this.spanSize = spanSize
        this.content = content
    }

    constructor(itemType: Int, spanSize: Int) {
        this.itemType = itemType
        this.spanSize = spanSize
    }

    var content: String? = null

    override fun getItemType(): Int {
        return itemType
    }

    companion object {
        val TEXT = 1
        val IMG = 2
        val IMG_TEXT = 3
        val TEXT_SPAN_SIZE = 3
        val IMG_SPAN_SIZE = 1
        val IMG_TEXT_SPAN_SIZE = 4
        val IMG_TEXT_SPAN_SIZE_MIN = 2
    }
}
